message/metadata.go

```diff
@@ -1,7 +1,14 @@
 package message
 
+import "encoding/json"
+
 type Metadata map[string]string
 
+type metadataItem struct {
```

Comment:
but why not to just use native map marshal?

Reply:
yes 4944c4db8425df59013ee6bc96ebf7d6fcfbad36